# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1763589405 -3600
# Parent  524eed0051b5dfc71d45dbbd75ceff89d5f45c40
Misc: Introduce pref gloda.facetview.max_messages_to_show

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -697,8 +697,9 @@ pref("gloda.facetview.sortby", 2);
 // 0 - Open facet view
 // 1 - Open facet view and list view
 // 2 - Open list view only (closing facet view)
 pref("gloda.list.immediate", 0);
+pref("gloda.facetview.max_messages_to_show", 10);
 
 // Enable gloda by default!
 pref("mailnews.database.global.indexer.enabled", true);
 // Limit the number of gloda message results
diff --git a/mail/base/content/glodaFacetView.js b/mail/base/content/glodaFacetView.js
--- a/mail/base/content/glodaFacetView.js
+++ b/mail/base/content/glodaFacetView.js
@@ -605,9 +605,14 @@ var FacetContext = {
    */
   planLayout() {
     // XXX arbitrary!
     this.maxDisplayRows = 8;
-    this.maxMessagesToShow = 10;
+    this.maxMessagesToShow = Services.prefs.getIntPref(
+      "gloda.facetview.max_messages_to_show",
+      10
+    );
+    this.maxMessagesToShow = Math.min(this.maxMessagesToShow, 1000);
+    this.maxMessagesToShow = Math.max(this.maxMessagesToShow, 10);
   },
 
   /**
    * Clean up the UI in preparation for a new query to come in.
